src/main/java/uk/ac/ic/wlgitbridge/git/handler/WLRepositoryResolver.java

```diff
@@ -97,6 +98,8 @@ public Repository open(
         } catch (ServiceNotEnabledException e) {
             cannot occur
             */
+        } catch (MissingRepositoryException e) {
+            throw new ServiceMayNotContinueException(String.join("\n", e.getDescriptionLines()), e);
```

Comment:
It took me a while to get my head around this, but hopefully I have now done so.

- [`ServiceMayNotContinueException `](http://download.eclipse.org/jgit/docs/jgit-2.0.0.201206130900-r/apidocs/org/eclipse/jgit/transport/ServiceMayNotContinueException.html) is an internal JGit exception, and this method is implementing an internal JGit interface, so we can't throw anything else. The docs say that the message for `ServiceMayNotContinueException` is intended for the end user, so that's good (matches what we are doing).
- `MissingRepositoryException` is one of ours, and it derives from `GitUserException` (via `SnapshotAPIException`, though I am not entirely sure that is right, because it doesn't do anything in the `fromJSON` method it needs to have because of `SnapshotAPIException`).
- `GitUserException` is sent to the user as a message and then a series of lines (the description), as part of the receive-pack protocol, in particular via `handleSnapshotPostException` in `WritelatexPutHook`. (It also gets sent as a series of lines from the `Oauth2Filter`, this time without the message, I think because git adds a helpful message in this case anyway.)
- This try-catch block handles `GitUserException` below, but it only passes back the message, not the description lines. This seems a bit inconsistent; I don't see a particular reason not to send back a multi-line message in this case.

So... another option here would be to change the `GitUserException` handler so that it also sends back the lines. That would also potentially improve the error messages in some other cases. WDYT?